src/formatDate.js

```diff
@@ -8,7 +8,70 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+  const dateParts = date.split(oldSeparator);
+  let year = '';
+  let month = '';
+  let day = '';
+  const formatDateParts = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i][0]) {
+      case 'Y':
+        year = dateParts[i];
+        break;
+      case 'M':
+        month = dateParts[i];
+        break;
+      case 'D':
+        day = dateParts[i];
+        break;
+      default:
+        throw new Error('not correct format');
+    }
```

Comment:
The error message 'not correct format' is not very informative. It would be beneficial to specify which format (fromFormat or toFormat) is incorrect and provide more details about what is expected. This can help the user of the function understand what they did wrong more quickly.